Pycloud Deployer/pycloud_deployer.py

```diff
@@ -0,0 +1,629 @@
+            subnet_name (str): Name of the subnet
+            nsg_name (str): Name of the NSG group
+            webserver_name (str): Prefix for webserver's name
+            dbserver_name (str): Name of DBserver
+        """
+        self.create_infrastructure(group_name, vnet_name, 
+                                   subnet_name, nsg_name, True)
+
+        self.webserver_provisioner.create_webserver(group_name, vnet_name,
+                                                    subnet_name, webserver_name)
+
+        self.dbserver_provisioner.create_dbserver(group_name, vnet_name, 
+                                                  subnet_name, dbserver_name) 
+
+    def tier3(self, group_name, vnet_name, subnet_name, nsg_name,
+              webserver_name, dbserver_name, number_of_webservers):
+        """Deploy a 3-tier infrastructure of webservers, dbservers and load balancers
+        
+        Args:
+            group_name (str): Name of the resource group
+            vnet_name (str): Name of the VNet
+            subnet_name (str): Name of the subnet
+            nsg_name (str): Name of the NSG group
+            webserver_name (str): Prefix for webserver's name
+            dbserver_name (str): Name of DBserver
+            number_of_webservers (int): Number of webservers to deploy
+        """
+        self.create_infrastructure(group_name, vnet_name, 
+                                    subnet_name, nsg_name, False)
+
+        # Rather than taking name inputs for each server, add server numbers to the
+        # webserver name to differentiate them
+        for server_number in range(number_of_webservers):
+            self.webserver_provisioner.create_webserver(group_name, vnet_name, subnet_name,
+                                                        webserver_name, server_number)
+        
+        self.dbserver_provisioner.create_dbserver(group_name, vnet_name, 
+                                                  subnet_name, dbserver_name)
+
+        self.lb_provisioner.create_lb(group_name)
+
+        self.lb_provisioner.add_webserver_to_backend(group_name, vnet_name, subnet_name,
+                                                     webserver_name, number_of_webservers)
+
+    def create_resource_group(self, group_name):
+        """Create resource group
+
+        Args:
+            group_name (str): Name of resource group
+        
+        Returns:
+            Resource group name
+        """
+        logger.info("Creating resoure group")
+
+        # Check if resource group exists and prompts the user for a name change, 
+        # if neccessary. Although an existing resource group can be used, it is 
+        # just to ensure the user knows about it.
+        if self.rss_client.resource_groups.check_existence(group_name):
+            logger.error(f"The resource group {group_name} already exists.")
+            
+            choice = input("Press Y to change group name or any other key to continue: ")
```

Comment:
consider making this more automatic instead of relying on user confirmation/input